src/components/home/HomePage.js

```diff
@@ -1,15 +1,64 @@
 import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
 import { CardList } from '../Card';
 import mockArticles from '../../../mockdata/articles';
 import PopularPosts from './PopularPosts';
+import { fetchCategory } from '../../actions/articleActions';
+
+export const CardListArray = (props) => {
+  const { articlesByCategory } = props;
+  const categories = Object.entries(articlesByCategory);
+
+  // check if the categories are still loading
+  const fetchedCategories = categories.filter(categoryArray => (categoryArray[1].articles));
+  const failedCategories = categories.filter(categoryArray => (categoryArray[1].error));
+  const loading = categories.length > (fetchedCategories.length + failedCategories.length);
+  if (loading) {
+    return <i className="fas fa-spinner fa-3x fa-spin loading-icon" />;
+  }
 
-class HomePage extends Component {
-  state = {
-    articles: mockArticles
-  };
+  return categories.map((category) => {
+    if (category[1].articles) {
+      return (
+        <CardList
+        key={category[0]}
+        className="home-articles-cont"
+        title={category[0]}
+        article={category[1].articles.slice(0, 6)} />
+      );
+    }
+    return (
+      <h5 key={category[0]}>
+        <strong>{category[0]}</strong>
+        <br />
+        {category[1].error.message}
+      </h5>
+    );
+  });
+};
+
+export class HomePageComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      articles: mockArticles
```

Comment:
mockArticles? are we fetching from database or from mockData?

Reply:
This is used to populate the popular posts sidebar. There is currently no way to fetch the most popular article from the database. or no way to even know which articles are the most popular. So I left this as mock data.

The only alternative I see here is to replace this sidebar with most recent posts. I was planning to bring this up in our next sync/TTL interaction